src/notebook/notebook/actions.ts

```diff
@@ -880,6 +883,34 @@ namespace Private {
   }
 
   /**
+   * Handle payloads from an execute reply.
+   *
+   * #### Notes
+   * Payloads are deprecated and there are no official interfaces for them in
+   * the kernel type definitions.
+   * See [Payloads (DEPRECATED)](http://jupyter-client.readthedocs.io/en/latest/messaging.html#payloads-deprecated).
+   */
+  function handlePayload(content: KernelMessage.IExecuteOkReply, parent: Notebook, child: BaseCellWidget) {
+    let setNextInput = content.payload.filter(i => {
+      return (i as any).source === 'set_next_input';
+    })[0];
+    if (setNextInput) {
+      let text = (setNextInput as any).text;
+      let replace = (setNextInput as any).replace;
+      if (replace) {
+        child.model.source = text;
+      } else {
+        // Create a new code cell and add as the next cell.
+        let cell = parent.model.factory.createCodeCell();
+        cell.source = text;
+        let cells = parent.model.cells;
+        let i = cells.indexOf(child.model);
```

Comment:
This is a utility function that could be called by whomever, whenever :). Our convention is to call it from the running cell....